src/routes/dashboard/containers/DashboardContainer.js

```diff
@@ -54,12 +56,36 @@ const mapStateToProps = (state: StateType, ownProps) => {
   }
 }
 
+const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>, ownProps: OwnPropsType) => ({
+  navigateToDashboard: createNavigateToCategory('Dashboard', dispatch, ownProps.navigation),
+  navigateToCategory: createNavigateToCategory('Categories', dispatch, ownProps.navigation),
+  navigateToEvent: createNavigateToEvent(dispatch, ownProps.navigation),
+  navigateToIntegreatUrl: createNavigateToIntegreatUrl(dispatch, ownProps.navigation),
+  changeUnavailableLanguage: (city: string, newLanguage: string) => {
+    const switchContentLanguage: SwitchContentLanguageActionType = {
+      type: 'SWITCH_CONTENT_LANGUAGE',
+      params: {
+        city,
+        newLanguage
+      }
+    }
+    dispatch(switchContentLanguage)
+    createNavigateToCategory('Dashboard', dispatch, ownProps.navigation)({
+      cityCode: city,
+      language: newLanguage,
+      path: `/${city}/${newLanguage}`,
+      forceUpdate: false,
+      key: ownProps.navigation.getParam('key')
+    })
+  }
+})
+
 export default compose([
-  withRouteCleaner,
   connect(mapStateToProps, mapDispatchToProps),
-  // TODO NATIVE-112 Show errors, maybe use withError and pass more props
-  branch(props => props.error, renderComponent(Failure)),
-  branch(props => props.invalidLanguage, renderComponent(LanguageNotAvailableContainer)),
+  withRouteCleaner,
+  withError,
+  withLanguageNotAvailable,
+  withLoading,
```

Comment:
not sure about these `withLoading`s as well... As of now this is handled in the `Dashboard` directly using the ScrollView in the RefreshControl. This allows the pull-to-refresh animation not to be interrupted by changing the view to the ActivityIndicator.
If you want to keep them, you would still need to touch Dashboard.js Events.js CategoriesScrollView.js and remove the loading-logic there.

Reply:
Apparently hooks are solving issues like this. The library recompose was abandoned because of hooks.

So maybe we can let it stay like this and refactor this in the future.

Reply:
I guess, my comment is still valid and needs to be resolved though...
Either remove the `withLanding`s again or remove the RefreshControl loading-logic from the routes

Reply:
Yeah right. Will do that